extensions/amp-bind/0.1/bind-expr-defines.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A single node in the AST of a `BindExpression`.
+ */
+export class AstNode {
+  /**
+   * @param {AstNodeType} type
+   * @param {?Array<AstNode>} args
+   * @param {(AstNodeValue|undefined)=} opt_value
+   */
+  constructor(type, args, opt_value) {
+    /** @const {AstNodeType} */
+    this.type = type;
+
+    /** @const {?Array<AstNode>} */
+    this.args = args;
+
+    /** @const {(AstNodeValue|undefined)} */
+    this.value = opt_value;
+  }
+}
+
+/**
+ * Type of a node in the AST of a `BindExpression`.
+ * @enum {string}
```

Comment:
I'd use a number. Strings mostly make sense if you want to compare them to external input.